dynamiq/memory/backends/qdrant.py

```diff
@@ -31,36 +31,91 @@ class Qdrant(MemoryBackend):
+                dimension=self.vector_store["dimension"],
+                on_disk=self.vector_store["on_disk"],
+                use_sparse_embeddings=self.vector_store["use_sparse_embeddings"],
+                sparse_idf=self.vector_store.get("sparse_idf", False),
+                metric=self.vector_store["metric"],
+                return_embedding=self.vector_store["return_embedding"],
+                write_batch_size=self.vector_store["write_batch_size"],
+                scroll_size=self.vector_store["scroll_size"],
+                content_key=self.vector_store["content_key"],
+                create_if_not_exist=self.vector_store.get("create_if_not_exist", True),
+                recreate_index=self.vector_store.get("recreate_index", False),
+                payload_fields_to_index=self.vector_store.get("payload_fields_to_index"),
+            )
+        elif not self.vector_store:
             self.vector_store = QdrantVectorStore(
                 connection=self.connection,
                 index_name=self.index_name,
-                create_if_not_exist=self.create_if_not_exist,
                 metric=self.metric,
                 on_disk=self.on_disk,
+                create_if_not_exist=self.create_if_not_exist,
                 recreate_index=self.recreate_index,
             )
+
         self._client = self.vector_store._client
         if not self._client:
             raise QdrantError("Failed to initialize Qdrant client")
 
-            # Ensure collection exists before any operations
-            if self.create_if_not_exist and not self._client.collection_exists(self.index_name):
-                self.vector_store._set_up_collection(
-                    collection_name=self.index_name,
-                    embedding_dim=1536,
-                    create_if_not_exist=True,
-                    recreate_collection=self.recreate_index,
-                    similarity=self.metric,
-                    use_sparse_embeddings=False,
-                    sparse_idf=False,
-                    on_disk=self.on_disk,
-                )
+    @model_validator(mode="after")
+    def validate_vector_store(self):
+        """Validate and initialize vector store if needed."""
+        if isinstance(self.vector_store, dict):
+            try:
+                self.vector_store = QdrantVectorStore(connection=self.connection, **self.vector_store)
+            except Exception as e:
+                raise ValueError(f"Failed to initialize vector store from dict: {e}")
+        return self
+
+    def to_dict(self, include_secure_params: bool = False, **kwargs) -> dict:
+        """Converts the instance to a dictionary."""
+        kwargs.pop("include_secure_params", None)
+        data = super().to_dict(**kwargs)
+
+        # Add embedder data
+        data["embedder"] = self.embedder.to_dict(include_secure_params=include_secure_params, **kwargs)
+
+        # Add vector store data if exists
+        if self.vector_store:
```

Comment:
We shouldn't handle clients dump for now as this is pretty tricky and buggy. We are skiping this for all our nodes, so let's skip for memory as well.